diff -urN Slic3r-1.2.9.org/xs/Build.PL Slic3r-1.2.9/xs/Build.PL
--- Slic3r-1.2.9.org/xs/Build.PL	2015-06-17 10:38:28.000000000 +0200
+++ Slic3r-1.2.9/xs/Build.PL	2015-06-21 16:51:02.208189103 +0200
@@ -14,6 +14,18 @@
     # only on newer GCCs: -ftemplate-backtrace-limit=0
     push @cflags, qw(-DSLIC3R_DEBUG -g);
 }
+my @libs;
+if ($ENV{SYSTEM_ADMESH}) {
+    push @libs, qw(-ladmesh);
+}
+if ($ENV{SYSTEM_POLYCLIPPING}) {
+    push @libs, qw(-polyclipping);
+}
+if ($ENV{SYSTEM_POLY2TRI}) {
+    push @cflags, qw(-I/usr/include/poly2tri);
+    push @libs, qw(-lpoly2tri);
+}
+
 if (ExtUtils::CppGuess->new->is_gcc) {
     # check whether we're dealing with a buggy GCC version
     # see https://github.com/alexrj/Slic3r/issues/1965
@@ -41,6 +53,7 @@
         Module::Build::WithXSpp     0.13
     )},
     extra_compiler_flags => \@cflags,
+    extra_linker_flags => \@libs,
     
     # Provides extra C typemaps that are auto-merged
     extra_typemap_modules => {
