diff --git a/lib/Slic3r/GUI/AboutDialog.pm b/lib/Slic3r/GUI/AboutDialog.pm
index 6fc83b6..78f89ef 100644
--- a/lib/Slic3r/GUI/AboutDialog.pm
+++ b/lib/Slic3r/GUI/AboutDialog.pm
@@ -66,7 +66,7 @@ sub new {
     $vsizer->Add($html, 1, wxEXPAND | wxALIGN_LEFT | wxRIGHT | wxBOTTOM, 20);
     EVT_HTML_LINK_CLICKED($self, $html, \&link_clicked);
     
-    my $buttons = $self->CreateStdDialogButtonSizer(wxCLOSE);
+    my $buttons = $self->CreateStdDialogButtonSizer(wxOK);
     $self->SetEscapeId(wxID_CLOSE);
     EVT_BUTTON($self, wxID_CLOSE, sub {
         $self->EndModal(wxID_CLOSE);
