--- Slic3r-1.2.6/lib/Slic3r.pm~	2015-02-01 15:47:29.000000000 +0100
+++ Slic3r-1.2.6/lib/Slic3r.pm	2015-05-15 11:24:02.024661971 +0200
@@ -26,11 +26,8 @@
     $have_threads = 0 if $Moo::VERSION == 1.003000;
 }
 
-warn "Running Slic3r under Perl 5.16 is not supported nor recommended\n"
-    if $^V == v5.16;
-
 use FindBin;
-our $var = "$FindBin::Bin/var";
+our $var = "/usr/share/slic3r";
 
 use Encode;
 use Encode::Locale;
